extensions/amp-ad/0.1/amp-ad-ui.js

```diff
@@ -117,6 +122,36 @@ export class AmpAdUIHandler {
   }
 
   /**
+   * See BaseElement method.
+   */
+  createPlaceholderCallback() {
+    if (!isExperimentOn(this.baseInstance_.win, UX_EXPERIMENT)) {
+      return null;
+    }
+    const placeholder = createElementWithAttributes(this.doc_, 'div', {
+      'placeholder': '',
+    });
+    placeholder.appendChild(this.createDefaultHolder_());
+    this.baseInstance_.element.appendChild(placeholder);
+    return placeholder;
+  }
+
+  /**
+   * Create the same holder for placeholder and fallback
+   * @return {!Element}
+   */
+  createDefaultHolder_() {
+    const adTagHolder = this.doc_.createElement('div');
```

Comment:
instead of creating the DOM manually, how about use innerHTML ? Which is more readable and [faster](http://www.quirksmode.org/dom/innerhtml.html).